rules/code-quality/src/Rector/Ternary/SwitchNegatedTernaryRector.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\CodeQuality\Rector\Ternary;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\BooleanNot;
+use PhpParser\Node\Expr\Ternary;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\CodeQuality\Tests\Rector\Ternary\SwitchNegatedTernaryRector\SwitchNegatedTernaryRectorTest
+ */
+final class SwitchNegatedTernaryRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Switch negated ternary condition rector',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run(bool $upper, string $name)
+    {
+        return ! $upper
+            ? $name
+            : strtoupper($name);
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+class SomeClass
+{
+    public function run(bool $upper, string $name)
+    {
+        return $upper
+            ? strtoupper($name)
+            : $name;
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]);
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getNodeTypes(): array
+    {
+        return [Ternary::class];
+    }
+
+    /**
+     * @param Ternary $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if (! $node->cond instanceof BooleanNot) {
+            return null;
+        }
+
+        if ($node->if === null) {
+            return null;
+        }
+
+        $node->cond = $node->cond->expr;
+        $if = $node->if;
+        $node->if = $node->else;
+
+        $node->else = $if;
```

Comment:
Here this trick can be used:

```suggestion
        [$node->if, $node->else] = [$node->else, $node->if];
```

Reply:
updated.